rest-service/manager_rest/rest/resources_v3_1/nodes.py

```diff
@@ -0,0 +1,75 @@
+from ..resources_v3 import Nodes as v3_Nodes
+
+from manager_rest.rest import rest_utils
+from manager_rest.security.authorization import authorize
+from manager_rest.storage import get_storage_manager, models
+from manager_rest.rest import rest_decorators
+
+
+class Nodes(v3_Nodes):
+    @authorize('node_create')
+    @rest_decorators.marshal_with(models.Node)
+    def post(self):
+        request_dict = rest_utils.get_json_and_verify_params({
+            'nodes': {'type': list},
+        })
+        sm = get_storage_manager()
+        raw_nodes = request_dict['nodes']
+        created = []
+        with sm.transaction():
+            deployment_id = self._deployment_id_from_nodes(raw_nodes)
+            deployment = sm.get(models.Deployment, deployment_id)
+            for raw_node in raw_nodes:
+                node = self._node_from_raw_node(raw_node)
+                node.set_deployment(deployment)
+                sm.put(node)
+                created.append(node)
+        return created, 200
+
+    def _deployment_id_from_nodes(self, raw_nodes):
```

Comment:
Isn't this roughly equivalent to just doing raw_nodes[0]['deployment_id'], but with an effective extra implicit assertion that all raw nodes contain a deployment ID?

Reply:
it is, you think there's a better way to do this? i think i want to have the check in some form

Reply:
I think making it more explicit might be better, e.g.
deployment_id = raw_nodes[0]['deployment_id']
for node in raw_nodes:
-indent-eaten-by-github-    if node['deployment_id'] != deployment_id: raise BadThingHappened('I feel sad')
return deployment_id

Reply:
OK, I see what you did with this later- that makes it make more sense, so ignore this.

Reply:
that makes for a worse error message, but OK, it's faster